pallets/capacity/src/types.rs

```diff
@@ -46,6 +51,46 @@ impl<T: Config> StakingAccountDetails<T> {
 		let available_staking_balance = account_balance.saturating_sub(self.total);
 		available_staking_balance.min(proposed_amount)
 	}
+
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	///  tmp fn for testing only
+	/// set unlock chunks with (balance, thaw_at).  does not check that the unlock chunks
+	/// don't exceed total.
+	/// returns true on success, false on failure (?)
+	// TODO: remove this when finished and use production fn
+	pub fn set_unlock_chunks(&mut self, chunks: &Vec<(u32, u32)>) -> bool {
+		let result: Vec<UnlockChunk<BalanceOf<T>, <T>::BlockNumber>> = chunks
+			.into_iter()
+			.map(|chunk| UnlockChunk { value: chunk.0.into(), thaw_at: chunk.1.into() })
+			.collect();
+		self.unlocking = BoundedVec::try_from(result).unwrap();
+		self.unlocking.len() == chunks.len()
+	}
+
+	/// deletes thawed chunks, updates `total`, Caller is responsible for updating free/locked
+	/// balance on the token account.
+	/// Returns: the total amount reaped from `unlocking`
+	pub fn reap_thawed(&mut self, current_block: <T>::BlockNumber) -> BalanceOf<T> {
+		let mut total_reaped: BalanceOf<T> = 0u32.into();
+		self.unlocking.retain(|chunk| {
+			if current_block.ge(&chunk.thaw_at) {
+				total_reaped = total_reaped + chunk.value;
+				match self.total.checked_sub(&chunk.value) {
+					Some(new_total) => self.total = new_total,
+					None => {
```

Comment:
Not requesting a change, but I just wanted to run a thought. Should we do a log or throw an underflow error instead of printing? Since errors are included in a block, we can look for those errors.

ArithmeticError::Underflow

Reply:
I'm not sure what you mean by "do a log...instead of printing" since the warn macro is a log output.  When we discussed this before, I mentioned how you can't return an error from `retain` iterator. We decided that it was an extremely unlikely edge case, that it's more or less harmless to just delete the chunk in the case of an underflow, and to log it in case it happens so it can be debugged.  So I chose warn over debug or info.

Reply:
I was thinking in general printing vs. throwing an error. I did not mean to confuse you. I used to log and print interchangeably. I am still figuring out if there is a rule of thumb to when to print instead of throwing an error. 
Again, I'm not asking for a change.